test/test_var_view.py

```diff
@@ -32,3 +41,309 @@ def test_get_stringifier():
 
             s = strifier(value)
             assert isinstance(s, text_type)
+
+
+class FrameVarInfoForTesting(FrameVarInfo):
+    def get_inspect_info(self, id_path, read_only):
+        iinfo = super(FrameVarInfoForTesting, self).get_inspect_info(
+            id_path, read_only)
+        iinfo.access_level = "private"
+        iinfo.display_type = "repr"
+        iinfo.show_detail = True
+        iinfo.show_methods = True
+        return iinfo
+
+
+class Reasonable(object):
+    def __init__(self):
+        self.x = 42
+
+    def bar(self):
+        return True
+
+    @property
+    def red(self):
+        return "red"
+
+    @classmethod
+    def blue(cls):
+        return "blue"
+
+    @staticmethod
+    def green():
+        return "green"
+
+    def _private(self):
+        return "shh"
+
+    def __magicsomething__(self):
+        return "amazing"
+
+
+class SetWithOverridenBool(set):
+    def __init__(self, iterable, truthy=True):
+        super(SetWithOverridenBool, self).__init__(iterable)
+        self.truthy = truthy
+
+    def __bool__(self):
+        return self.truthy
+
+
+def containerlike_class_generator():
+    methods = set([
+        "__contains__",
+        "__getitem__",
+        "__iter__",
+        "__len__",
+        "__reversed__",
+        "count",
+        "get",
+        "index",
+        "items",
+        "keys",
+        "values",
+    ])
+
+    # Deliberately starting from 0
+    for r in range(0, len(methods) + 1):
+        for selected_methods in sorted(
+                map(sorted, itertools.combinations(methods, r))):
```

Comment:
Lol. :)

How many classes does this generate? How long do the resulting tests take?

Reply:
1. Lots :P
2. Not too long, at least for me (6 cores, 12 threads @ 3.2 GHz, 16GB RAM):
```
$ time pytest test/test_var_view.py
===================================================================================== test session starts =====================================================================================
platform linux -- Python 3.8.5, pytest-6.2.1, py-1.10.0, pluggy-0.13.1
rootdir: ---
collected 9 items

test/test_var_view.py .........

====================================================================================== 9 passed in 0.81s ======================================================================================

real    0m1.007s
user    0m0.959s
sys     0m0.048s
```

Reply:
More accurately: it generates 2048 classes.

Reply:
Oh OK. That's not too bad then. Thanks for checking!